src/include/miopen/fusion/solvers.hpp

```diff
@@ -0,0 +1,206 @@
+    virtual boost::any GetDefaultPerformanceConfig(const FusionContext& ctx, int) const = 0;
+
+    /// Should return false if performance config is wrong for a problem.
+    /// Main use is validation of values read from the perf db.
+    virtual bool IsValidPerformanceConfig(const FusionContext& ctx,
+                                          const boost::any& config) const = 0;
+
+    /// Search
+    ///
+    /// The int parameter is needed only to not change the name of the
+    /// function in the derived class. Function declarations that differ
+    /// only by its return type cannot be overloaded.
+    virtual boost::any
+    Search(const FusionContext& ctx, const AnyInvokeParams& invoke_ctx, int) const = 0;
+
+    /// Tunable solvers provide a GetSolution that takes a Context and PerformanceConfig
+    virtual ConvSolution GetSolution(const FusionContext& ctx, const boost::any& config) const = 0;
+};
+
+template <class PerformanceConfig>
+struct FusionTunableSolver : FusionTunableSolverBase
+{
+    virtual PerformanceConfig GetDefaultPerformanceConfig(const FusionContext&) const           = 0;
+    virtual bool IsValidPerformanceConfig(const FusionContext&, const PerformanceConfig&) const = 0;
+    virtual PerformanceConfig Search(const FusionContext&, const AnyInvokeParams&) const        = 0;
+    virtual ConvSolution GetSolution(const FusionContext&, const PerformanceConfig&) const      = 0;
+
+    boost::any GetDefaultPerformanceConfig(const FusionContext& ctx, int) const final
+    {
+        return GetDefaultPerformanceConfig(ctx);
+    }
+
+    bool IsValidPerformanceConfig(const FusionContext& ctx, const boost::any& config) const final
+    {
+        return IsValidPerformanceConfig(ctx, boost::any_cast<const PerformanceConfig&>(config));
+    }
+
+    boost::any Search(const FusionContext& ctx, const AnyInvokeParams& invoke_ctx, int) const final
+    {
+        return Search(ctx, invoke_ctx);
+    }
+
+    ConvSolution GetSolution(const FusionContext& ctx, const boost::any& config) const final
+    {
+        return GetSolution(ctx, boost::any_cast<const PerformanceConfig&>(config));
+    }
+    bool IsDynamic() const override { return false; }
+};
+
+struct PerformanceConfigConvBiasActivAsm1x1U : PerformanceConfigConvAsm1x1U
+{
+    PerformanceConfigConvBiasActivAsm1x1U(const bool spare) : PerformanceConfigConvAsm1x1U(spare) {}
+    PerformanceConfigConvBiasActivAsm1x1U()
+        : PerformanceConfigConvAsm1x1U(-1, -1, -1, -1, -1, -1, -1, -1, false)
+    {
+    }
+    void HeuristicInit(const FusionContext& ctx);
+    bool SetNextValue(const FusionContext& problem);
+    bool IsValid(const FusionContext& problem) const;
+};
+
+struct ConvBiasActivAsm1x1U : FusionTunableSolver<PerformanceConfigConvBiasActivAsm1x1U>
+{
+    const std::string& SolverDbId() const override { return GetSolverDbId<ConvBiasActivAsm1x1U>(); }
+
+    bool IsApplicable(const FusionContext& problem) const override;
+    ConvSolution
```

Comment:
@JehandadKhan Probably `GetWorkspaceSize()` and `MayNeedWorkspace()` are missing

Reply:
@averinevg If WS is not needed, then the solver can safely derive them from `SolverMixin<FusionContext>`, right?

Reply:
@atamazov Yes, but this solver comes from `src/include/miopen/solver.hpp`, and these functions are there

Reply:
The Fusion solvers are derived from `SolverMixin<FusionContext>` and inherit both of these methods since it derives from `SolverBase`

Reply:
@JehandadKhan `ConvBiasActivAsm1x1U` uses `GetSolution` from `ConvAsm1x1U`, and `ConvAsm1x1U` has these methods